fix: fixed the notify triager workflow

sambhavgupta0705 opened the pull request:
The current workflow has an error in which it calls the code reviewers also when the changes are only in the md files 

Fixed that by adding a check

fixes : #3214 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced pull request handling workflow with improved commit message extraction and file change counting.
	- Streamlined reviewer assignment logic based on the number of changed files.

- **Bug Fixes**
	- Updated conditions for adding reviewers to ensure accurate processing based on file types.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sambhavgupta0705 committed:
fix: fixed the notify triager workflow

sambhavgupta0705 committed:
Merge branch 'master' of github.com:sambhavgupta0705/asyncapi_website into bounty2

sambhavgupta0705 commented:
Tested the workflow here 
https://github.com/sambhavgupta0705/asyncapi_website/pull/5
cc: @akshatnema @anshgoyalevil @devilkiller-ag

sambhavgupta0705 committed:
Update notify-triager.yml

aeworxet commented:
@asyncapi/bounty_team

anshgoyalevil commented on .github/workflows/notify-triager.yml:
```diff
@@ -68,7 +69,7 @@ jobs:
 
       - name: Add Reviewers for code files
-        if: steps.check-merge-branch.outputs.isMergeCommit == 'false' && steps.non-md-pr-changes.outputs.any_changed == 'true'
+        if: ${{ steps.check-merge-branch.outputs.isMergeCommit == 'false' && steps.changed-files.outputs.all_changed_files_count != steps.md-pr-changes.outputs.all_changed_files_count }}
```
What if the `changed-files.outputs.all_changed_files_count` is equal to `steps.md-pr-changes.outputs.all_changed_files_count`? 
Can you please explain exactly what that check means?

sambhavgupta0705 replied:
For this we are checking for non md files
If the count is not equal that means there are some non md files also which needs to be reviewed by code reviewers

sambhavgupta0705 replied:
tj-actions doesn't have any mechanism for non md files so that's why I implemented this or else we need to check for many file extensions

anshgoyalevil replied:
Got it. Good catch

anshgoyalevil replied:
Though I am still curious about the ambigious behavior of this workflow. For example, examine this PR

https://github.com/asyncapi/website/pull/3245#issuecomment-2388259456

See how MD triagers are getting assigned as PR reviewers even though there was no merge commit or a MD file change

Another instance of this issue:
https://github.com/asyncapi/website/pull/3075#event-13815559910

sambhavgupta0705 replied:
https://github.com/asyncapi/website/actions/runs/11141478939/job/30962427072

I think this job was surpassed

sambhavgupta0705 committed:
Merge branch 'master' into bounty2

sambhavgupta0705 commented:
/update

sambhavgupta0705 committed:
Merge branch 'master' into bounty2

aeworxet commented:
Does this PR need to be reviewed also by @akshatnema before it can be merged?

JeelRajodiya committed:
Merge branch 'master' into bounty2

sambhavgupta0705 committed:
fix: using git diff

sambhavgupta0705 committed:
Merge branch 'bounty2' of github.com:sambhavgupta0705/asyncapi_website into bounty2

sambhavgupta0705 committed:
Merge branch 'master' into bounty2

sambhavgupta0705 committed:
triggering the workflow

sambhavgupta0705 committed:
trigger doc triagers

sambhavgupta0705 committed:
revert

sambhavgupta0705 commented:
@anshgoyalevil @akshatnema @aeworxet  I have updated the PR 
You can review it now

in this we have removed the tj-actions workflow and have written our own workflow with `git-diff`

sambhavgupta0705 committed:
fix

sambhavgupta0705 committed:
trigger doc reviewers

sambhavgupta0705 committed:
revert

sambhavgupta0705 committed:
fix

sambhavgupta0705 committed:
Update notify-triager.yml

anshgoyalevil commented:
/rtm
